// entry/entry.js
Object.extend = function(destination, source) {
    for (var property in source) {
        if (source.hasOwnProperty(property)) {
            destination[property] = source[property];
        }
    }
    return destination;
};

var entry = {
  GET: {},
  POST: {},
  DELETE: {},
  PUT: {}
}


var get = function(path, callback) {
  entry.GET[path] = callback;
};

var post = function(path, callback) {
  entry.POST[path] = callback;
};

var del = function(path, callback) {
  entry.DELETE[path] = callback;
};

var put = function(path, callback) {
  entry.PUT[path] = callback;
};

var set = function(method, path, callback) {
  entry[method][path] = callback;
}

var getPathCallback = function(method, path) {
  return entry[method.toUpperCase()][path];
};

var getCallbacks = function() {
  return entry;
}

var extend = function(callbacks) {
  entry = Object.extend(entry, callbacks);
}

exports.get = get;
exports.post = post;
exports.delete = del;
exports.put = put;
exports.getPathCallback = getPathCallback;
exports.callbacks = entry;
exports.getCallbacks = getCallbacks;
exports.extend = extend;
